src/paintingTools/brush.js

```diff
@@ -107,13 +107,265 @@ function onImageRendered (e) {
   }
 }
 
+// This method is for fill region of segmentation overlays
+
+function fill (eventData) {
+  const configuration = brush.getConfiguration();
+  const radius = configuration.radius;
+  const element = eventData.element;
+  const layer = external.cornerstone.getLayer(element, configuration.brushLayerId);
+  const { rows, columns } = layer.image;
+  let { x, y } = eventData.currentPoints.image;
+  const toolData = getToolState(element, TOOL_STATE_TOOL_TYPE);
+  const pixelData = toolData.data[0].pixelData;
+  const brushPixelValue = configuration.draw;
+
+  const dx = [0, 1, 0, -1];
+  const dy = [-1, 0, 1, 0];
+
+  if (x < 0 || x > columns ||
+    y < 0 || y > rows) {
+    return;
+  }
+
+  const getPixelIndex = (x, y) => (y * columns) + x;
+  const isSameColor = (x, y, color = brushPixelValue) => pixelData[getPixelIndex(x, y)] === color;
+  const draw = (x, y, brushPixelValue) => drawBrushPixels([[x, y]], pixelData, brushPixelValue, columns);
+  const drawRange = (range, color) => drawBrushPixels(range, pixelData, color, columns);
+
+  x = Math.round(x);
+  y = Math.round(y);
+
+  const find = (low, high, condition, base, isY = false) => {
```

Comment:
This function probably doesn't need to be an inner function. Pulling it out may make the logical flow of fill() easier to follow.

Reply:
I agree. find function is moved to fill.js